### PR TITLE
Add the fake initial snapshot to the snapshots.

### PR DESCRIPTION
**Pull request by nc6**
This is needed to allow the initial stake pools to continue producing
blocks into the second epoch.

HERE BE DRAGONS: this function is not suitable for production use!

### CODE REVIEW

**Comment by mrBliss** [comment]
> HERE BE DRAGONS: this function is not suitable for production use!

But now it *will be* used by any pure Shelley node. Or is that not "production"?

What are the consequences of using it?

The warning should really live in the code, not just in this commit message.

**Comment by nc6** [comment]
It will not be used unless `registerGenesisStaking` is invoked. But you're right, I should add this comment onto that function!

**Comment by mrBliss** [comment]
> It will not be used unless `registerGenesisStaking` is invoked. But you're right, I should add this comment onto that function!

But `registerGenesisStaking` *is currently* called in `protocolInfoShelley`.

**Comment by nc6** [comment]
Hmm, true. OK, I have added a comment that explains the use, but I think a better solution would be to make this optional in `ShelleyGenesis` and then avoid doing any manipulation. I've added an issue at https://github.com/input-output-hk/cardano-ledger-specs/issues/1633

**Comment by deepfire** [comment]
Ohh, but then pool block creation stops when the second epoch starts...

**Comment by deepfire** [comment]
Works just fine now!

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs [hunk 1]
```diff
@@ -253,6 +253,10 @@ protocolInfoShelley genesis initialNonce maxMajorPV protVer mbCredentials =
     --
     -- This function embodies a little more logic than ideal. We might want to
     -- move it into `cardano-ledger-specs.`
+    --
+    -- HERE BE DRAGONS! This function is intended to help in testing. It should
+    -- not be called with anything other than 'emptyGenesisStaking' in
+    -- production.
```
Comment (mrBliss) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:259]: So production /= Shelley testnets.

**Comment by nc6** [comment]
bors retry

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs (hunk 1 above)
Comment (nc6) [reply ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:260]: On shelley testnets it should be called with `emptyGenesisStaking`

**Pull request merged**